frontend/src/types/media.ts

```diff
@@ -103,10 +103,17 @@ export interface ImageDimensions {
 export type AspectRatio = "square" | "intrinsic"
 
 export const isDetail = {
-  audio: (media: Media): media is AudioDetail => {
-    return media.frontendMediaType === "audio"
+  audio: (media: AudioDetail | ImageDetail | null): media is AudioDetail => {
+    return !!media && media.frontendMediaType === "audio"
   },
-  image: (media: Media): media is ImageDetail => {
-    return media.frontendMediaType === "image"
+  image: (media: AudioDetail | ImageDetail | null): media is ImageDetail => {
+    return !!media && media.frontendMediaType === "image"
   },
 }
+
+export const isMediaDetail = <T extends SupportedMediaType>(
+  media: AudioDetail | ImageDetail | null,
+  mediaType: T
+): media is DetailFromMediaType<T> => {
+  return !!media && media.frontendMediaType === mediaType
+}
```

Comment:
Why do we need to use `AudioDetail | ImageDetail | null` rather than `Media | null`? `Media` inherently covers `AudioDetail` and `ImageDetail`.

Also, does the `isMediaDetail` function duplicate the `isDetail.*` functions? If we need the separated ones, the `isDetail` functions can be implemented with `isMediaDetail`:

```typescript
const isDetail = {
  audio: (media: Media | null) => isMediaDetail(media, "audio"),
  image: (media: Media | null) => isMediaDetail(media, "image"),
}
```

Reply:
I tried applying this suggestion, but then I get an error in the `VAllResultsGrid` component, which uses `isDetal.audio`/`isDetail.image` in the template. 
```
frontend/src/components/VSearchResultsGrid/VAllResultsGrid.vue(43,11): error TS2322: Type 'Media' is not assignable to type 'ImageDetail'.
  Types of property 'frontendMediaType' are incompatible.
    Type '"audio" | "image"' is not assignable to type '"image"'.
      Type '"audio"' is not assignable to type '"image"'.
frontend/src/stores/media/single-result.ts(47,9): error TS2322: Type '{ frontendMediaType: "image"; height?: number | undefined; width?: number | undefined; id: string; title: string; originalTitle: string; creator?: string | undefined; creator_url?: string | undefined; ... 21 more ...; isSensitive: boolean; } | { ...; } | null' is not assignable to type 'ImageDetail | null'.
  Type '{ frontendMediaType: "audio"; audio_set?: { title: string; foreign_landing_url: string; creator?: string | undefined; creator_url?: string | undefined; url?: string | undefined; filesize?: number | undefined; filetype?: string | undefined; } | undefined; ... 35 more ...; isSensitive: boolean; }' is not assignable to type 'ImageDetail'.
    Types of property 'frontendMediaType' are incompatible.
      Type '"audio"' is not assignable to type '"image"'.
```